WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -1043,7 +1046,13 @@ import WordPressComAnalytics
     /// are met.
     /// - The app must have a internet connection.
     /// - The current time must be greater than the last sync interval.
-    func syncIfAppropriate() {
+    /// - Autosync must not have already occurred.
+    ///
+    /// NOTE:  We want to avoid automatically syncing when the controller is created
+    /// via state restoration until the controller's view is in the window.
+    /// See: https://github.com/wordpress-mobile/WordPress-iOS/issues/6297
+    ///
+    func autoSyncIfAppropriate(viewWillAppear viewWillAppear: Bool) {
```

Comment:
I'm confused about this method and the changes. I might be getting this completely wrong, but here's my understanding so far:

- This is called on `viewWillAppear` and `configureControllerForTopic`, which in turn is called in `viewDidLoad` and when `readerTopic` is changed.
- The `viewWillAppear` argument and logic only starts to make sense once you know when it's going to be called. For that I suspect this is too tightly coupled with how it's being called, and perhaps calling for a refactor.
- Since it already has a check against `refreshInterval`, instead of a `didAutoSync` I think this needs to check if it's already syncing. `WPContentSyncHelper` already does this check internally so in theory you can call `autoSyncIfAppropriate` twice in a row and it'll only sync once.
- `didAutoSync` will prevent any further sync, even when it's necessary: go to Reader > Search, search for `food`, then search for anything else and it won't work.
- I believe the new check tries to prevent the sync from happening when invoked via `viewDidLoad` on UI restoration. I think sync should never be called from `viewDidLoad`. Can we just move `syncIfAppropriate` out of `configureControllerForTopic` and call it directly on `viewWillAppear` and `readerTopic.didSet`?